airflow/providers/postgres/hooks/postgres.py

```diff
@@ -184,7 +184,13 @@ def get_iam_token(self, conn: Connection) -> Tuple[str, str, int]:
             # Pull the custer-identifier from the beginning of the Redshift URL
             # ex. my-cluster.ccdre4hpd39h.us-east-1.redshift.amazonaws.com returns my-cluster
             cluster_identifier = conn.extra_dejson.get('cluster-identifier', conn.host.split('.')[0])
-            client = aws_hook.get_client_type('redshift')
+            session, endpoint_url = aws_hook._get_credentials()
+            client = session.client(
+                "redshift",
+                endpoint_url=endpoint_url,
+                config=aws_hook.config,
+                verify=aws_hook.verify,
+            )
```

Comment:
This changed the behavior for mocking

We could try change in `test_get_conn_rds_iam_redshift`
```
@mock.patch('airflow.providers.amazon.aws.hooks.base_aws.AwsBaseHook.get_client_type')
```
to
```
@mock.patch('airflow.providers.amazon.aws.hooks.base_aws.AwsBaseHook._get_credentials')
```

Reply:
I guess because we expect tuple it can't patch it. Not sure about how such thing should be handled in unit test framework. Any expert here?

Reply:
We can just tell the mock to return a tuple :) I’ll update.

Reply:
This should do it.